digitalout.py

```diff
@@ -85,47 +94,72 @@ def load_xml(self, node):
+                        self.close()
+                        msg = '\n DAQmxDO failed to close current task'
+                        raise HardwareError(self, task, message=msg)
 
-                self.task = nidaqmx.Task() # might be task.Task()
-                
-                # Create digital out virtual channel
-                self.task.do_channels.add_do_chan(
-                    lines=self.physicalChannels, 
-                    name_to_assign_to_lines="",
-                    line_grouping=LineGrouping.CHAN_FOR_ALL_LINES)
-                
-                # Setup timing. Use the onboard clock
-                self.task.timing.cfg_samp_clk_timing(
-                    rate=self.clockRate, 
-                    active_edge=Edge.RISING, # default
-                    sample_mode=AcquisitionType.FINITE, # default
-                    samps_per_chan=samplesPerChannel) 
+                    except DaqWarning as e:
+                        self.logger.warning(str(e.message))
+
+                try:
+                    self.task = nidaqmx.Task() # might be task.Task()
+                    
+                    # Create digital out virtual channel
+                    self.task.do_channels.add_do_chan(
+                        lines=self.physicalChannels, 
+                        name_to_assign_to_lines="",
+                        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES)
                     
-                # Optionally set up start trigger
-                if self.startTrigger.wait_for_start_trigger:
-                    self.task.start_trigger.cfg_dig_edge_start_trig(
-                        trigger_source=self.startTrigger.source,
-                        trigger_edge=self.startTrigger.edge)
-                                                        
-                # Write digital waveform 1 chan N samp
-                self.task.write(
-                    self.data, 
-                    auto_start=AUTO_START_UNSET, #default
-                    timeout=10.0) # default
+                    # Setup timing. Use the onboard clock
+                    self.task.timing.cfg_samp_clk_timing(
+                        rate=self.clockRate, 
+                        active_edge=Edge.RISING, # default
+                        sample_mode=AcquisitionType.FINITE, # default
+                        samps_per_chan=samplesPerChannel) 
+                        
+                    # Optionally set up start trigger
+                    if self.startTrigger.wait_for_start_trigger:
+                        self.task.start_trigger.cfg_dig_edge_start_trig(
+                            trigger_source=self.startTrigger.source,
+                            trigger_edge=self.startTrigger.edge)
+                                                            
+                    # Write digital waveform 1 chan N samp
+                    self.task.write(
+                        self.data, 
+                        auto_start=AUTO_START_UNSET, #default
```

Comment:
Where is the `AUTO_START_UNSET` constant defined?

Reply:
this is what I get for pasting from documentation and not using an IDE that catches these for me

Reply:
Using pycharm and scrolling through it yelling at me after I pulled the branch is how I generated 90% of this review. I suggest using pycharm.